### PR TITLE
Add confirmations when removing an entire node before setting disks to 0 weight

### PR DESCRIPTION
**Pull request by SuperSandro2000**

### CODE REVIEW

**Review by majewsky — CHANGES_REQUESTED** [review]
Since this PR is mostly refactorings and one new feature, my review is mostly stylistic nitpicks and one actual issue.

File: pkg/rules/rules.go [hunk 1]
```diff
@@ -95,11 +106,9 @@ func (ringRules RingRules) CalculateChanges(ring builderfile.RingInfo, ringFilen
 		commandQueue = append(commandQueue, ring.CommandSetOverload(ringFilename, ringRules.Overload))
 	}
 
-	var zoneIDs []uint64
-	for zoneID := range ringRules.Zones {
-		zoneIDs = append(zoneIDs, zoneID)
-	}
-	sort.Slice(zoneIDs, func(i, j int) bool { return zoneIDs[i] < zoneIDs[j] }) // for reproducibility in tests
+	zones := ringRules.getZones()
+	for _, zone := range zones {
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkg/rules/rules.go:126]: This is later used in line 189 and I didn't want to collect the infos twice.

**Review by majewsky — APPROVED** [review]

**Review by majewsky — CHANGES_REQUESTED** [review]
Sorry for the noise, I only reviewed the force-push diff and did not see that there were some unaddressed concerns.

**Comment by SuperSandro2000** [comment]
forgot to push that 😅

**Review by majewsky — APPROVED** [review]

**Pull request merged**